gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dnumslider_ttt2.lua

```diff
@@ -273,17 +278,58 @@ function PANEL:SetServerConVar(cvar)
 		end
 	end)
 
+	networkedVarTracker = networkedVarTracker % 1023 + 1
+	local myIdentifierString = "TTT2F1MenuServerConVarChangeCallback" .. tostring(networkedVarTracker)
+
 	local function OnServerConVarChangeCallback(conVarName, oldValue, newValue)
 		if not IsValid(self) then
-			cvars.RemoveChangeCallback(conVarName, "TTT2F1MenuServerConVarChangeCallback")
+			cvars.RemoveChangeCallback(conVarName, myIdentifierString)
 
 			return
 		end
 
 		self:SetValue(tonumber(newValue), true)
 	end
 
-	cvars.AddChangeCallback(cvar, OnServerConVarChangeCallback, "TTT2F1MenuServerConVarChangeCallback")
+	cvars.AddChangeCallback(cvar, OnServerConVarChangeCallback, myIdentifierString)
+end
+
+---
+-- @param table databaseInfo containing {name, itemName, key}
+-- @realm client
+function PANEL:SetDatabase(databaseInfo)
+	if not istable(databaseInfo) then return end
+
+	local name = databaseInfo.name
+	local itemName = databaseInfo.itemName
+	local key = databaseInfo.key
+
+	if not name or not itemName or not key then return end
+
+	self.databaseInfo = databaseInfo
+
+	database.GetValue(name, itemName, key, function(databaseExists, value)
+		if databaseExists then
+			self:SetValue(value, true)
+		end
+	end)
+
+	self:SetDefaultValue(database.GetDefaultValue(name, itemName, key))
+
+	networkedVarTracker = networkedVarTracker % 1023 + 1
+	local myIdentifierString = "TTT2F1MenuDatabaseChangeCallback" .. tostring(networkedVarTracker)
+
+	local function OnDatabaseChangeCallback(_name, _itemName, _key, oldValue, newValue)
+		if not IsValid(self) then
+			database.RemoveChangeCallback(name, itemName, key, myIdentifierString)
+
+			return
+		end
+
+		self:SetValue(newValue, true)
+	end
+
+	database.AddChangeCallback(name, itemName, key, OnDatabaseChangeCallback, myIdentifierString)
```

Comment:
I dont really like that we repeat this function all the time. 

We might move this to another file and define it once instead.
Is there a possibility to use a base class for such ttt2 panel features?

Reply:
Yeah it was my first effort to equalize all callbacks and SetValue. Didnt proceed to implement inheriting as its kinda confusing for panels. They dont all share an early parent. But with some effort it's definitely possible and useful.

Reply:
Right now, I would rather like to merge the PR and do this in another one, otherwise I have to always redo it for all those changes on master, which becomes really cumbersome. 
Not sure how long it takes me to properly handle the inheritance of these wildly assembled Panels.